src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+type ColorType = 'red' | 'green' | 'blue';
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+const PI = Math.PI;
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  color: ColorType;
+
+  shape: ShapeType = 'triangle';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: ColorType, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        // eslint-disable-next-line max-len
+        'The longest side of a triangle must be less than the sum of the other two.',
+      );
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
```

Comment:
According to the provided checklist, the `getArea` function should not accept any parameters. All necessary values should be available as object fields, which they are in this case. The function signature should be `getArea(): number` without parameters.